src/Toast.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import Fade from './Fade';
+import Header from './ToastHeader';
+import Body from './ToastBody';
+import { createBootstrapComponent } from './ThemeProvider';
+import ToastContext from './ToastContext';
+
+const propTypes = {
+  /**
+   * @default 'toast'
+   */
+  bsPrefix: PropTypes.string,
+
+  /**
+   * Apply a CSS fade transition to the toast
+   */
+  animation: PropTypes.bool,
+
+  /**
+   * Auto hide the toast
+   */
+  autohide: PropTypes.bool,
+
+  /**
+   * Delay hiding the toast (ms)
+   */
+  delay: PropTypes.number,
+
+  /**
+   * A Callback fired when the close button is clicked.
+   */
+  onClose: PropTypes.func,
+
+  /**
+   * When `true` The modal will show itself.
+   */
+  show: PropTypes.bool,
+
+  /**
+   * A `react-transition-group` Transition component used to animate the Toast on dismissal.
+   */
+  transition: PropTypes.elementType,
+
+  /** @ignore */
+  innerRef: PropTypes.any,
+};
+
+const defaultProps = {
+  animation: true,
+  autohide: false,
+  delay: 3000,
+  show: true,
+  transition: Fade,
+};
+
+class Toast extends React.Component {
+  constructor(props) {
+    super(props);
+    this.toastContext = {
+      onClose: () => this.props.onClose(),
+    };
+    const { autohide } = props;
+    if (autohide) {
+      this.scheduleAutohideTimeout(props.delay);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { autohide, delay, show } = this.props;
+    if (!this.autohideInterval && autohide && show !== prevProps.show) {
```

Comment:
```suggestion
    if (!this.autohideInterval && autohide && show && !prevProps.show) {
```

or make `scheduleAutohideTimeout` into `maybeScheduleAutohideTimeout` and put the check there

Reply:
not relevant anymore due the useEffect change.